website/static/js/nodesPrivacy.js

```diff
@@ -242,17 +242,22 @@ NodesPrivacyViewModel.prototype.confirmChanges =  function() {
         patchNodesPrivacy(nodesChanged).then(function () {
             self.onSetPrivacy(nodesChanged);
 
-            $osf.unblock();
             self.nodesChangedPublic([]);
             self.nodesChangedPrivate([]);
             self.page(self.WARNING);
             window.location.reload();
-        }).fail(function () {
+        }).fail(function (xhr) {
             $osf.unblock();
-            $osf.growl('Error', 'Unable to update project privacy');
+            var errorMessage = 'Unable to update project privacy';
+            if (xhr.responseJSON && xhr.responseJSON.errors) {
+                errorMessage = xhr.responseJSON.errors[0].detail;
+            }
+            $osf.growl('Problem changing privacy', errorMessage);
             Raven.captureMessage('Could not PATCH project settings.');
             self.clear();
-            window.location.reload();
+            $('#nodesPrivacy').modal('hide');
```

Comment:
Note to reviewer: I'm not 100% sure of the implication of this. I _believe_ the reload added [here](https://github.com/CenterForOpenScience/osf.io/commit/ff7bf4c884acb5d559604034f82d85d78215d2c4#diff-4f5e70a2bdc38e3df54149befd145224) might not be needed. I think it was in place for non-transactional bulk updates.